packages/sources/peregrine-fund-admin/src/transport/reserve.ts

```diff
@@ -0,0 +1,75 @@
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { BaseEndpointTypes } from '../endpoint/reserve'
+
+export interface ResponseSchema {
+  assetId: string
+  totalValue: number
+  currencyBase: string
+  accountIds: number[]
+  updateDateTime: string
+}
+
+// HttpTransport extends base types from endpoint and adds additional, Provider-specific types like 'RequestBody', which is the type of
+// request body (not the request to adapter, but the request that adapter sends to Data Provider), and 'ResponseBody' which is
+// the type of raw response from Data Provider
+export type HttpTransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: ResponseSchema
+  }
+}
+// HttpTransport is used to fetch and process data from a Provider using HTTP(S) protocol. It usually needs two methods
+// `prepareRequests` and `parseResponse`
+export const httpReserveTransport = new HttpTransport<HttpTransportTypes>({
+  // `prepareRequests` method receives request payloads sent to associated endpoint alongside adapter config(environment variables)
+  // and should return 'request information' to the Data Provider. Use this method to construct one or many requests, and the framework
+  // will send them to Data Provider
+  prepareRequests: (params, config) => {
+    return params.map((param) => {
+      return {
+        // `params` are parameters associated to this single request and will also be available in the 'parseResponse' method.
+        params: [param],
+        // `request` contains any valid axios request configuration
+        request: {
+          baseURL: config.DEFAULT_BASE_URL,
+          url: config.DEFAULT_RESERVE_URL,
```

Comment:
isn't this supposed to include the assetId as a path param the end?

Reply:
There are now API search query params, I just need to include the assetId as path a variable on the end. This is true for both end points.

i.e. https://www.myfoo.com/api/v1/nav/100, https://www.myfoo.com/api/v1/reserves/100
https://www.myfoo.com/api/v1/reserves/{assetId}

Reply:
I think i see the suggestion now will update with param injected into the endpoint string for the prepareRequest